src/vmm/benches/queue.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Benchmarking cases:
+//   * `Queue.pop`
+//   * `Queue.add_used`
+//   * `DescriptorChain.next_descriptor`
+
+use std::num::Wrapping;
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use vm_memory::GuestAddress;
+use vmm::devices::virtio::queue::{VIRTQ_DESC_F_NEXT, VIRTQ_DESC_F_WRITE};
+use vmm::devices::virtio::test_utils::VirtQueue;
+use vmm::utilities::test_utils::single_region_mem;
+
+/// Create one chain with n descriptors
+fn set_dtable_one_chain(rxq: &VirtQueue, n: usize) {
+    for i in 0..n {
+        rxq.dtable[i].set(
+            (2048 + (65562 / n) * i) as u64,
+            (65562 / n) as u32,
+            VIRTQ_DESC_F_WRITE | VIRTQ_DESC_F_NEXT,
+            (i + 1) as u16,
+        );
+        rxq.dtable[n - 1].flags.set(VIRTQ_DESC_F_WRITE);
+        rxq.dtable[n - 1].next.set(0);
```

Comment:
I think this should be outside of the loop

Reply:
fixed